sdk/libs-scala/jwt/src/main/scala/com/daml/jwt/JwksVerifier.scala

```diff
@@ -56,9 +57,23 @@ class JwksVerifier(
     .build()
 
   private[this] def getVerifier(keyId: String): Error \/ JwtVerifier = {
-    val jwk = http.get(keyId)
-    val publicKey = jwk.getPublicKey.asInstanceOf[RSAPublicKey]
-    RSA256Verifier(publicKey, jwtTimestampLeeway)
+    try {
+      val jwk = http.get(keyId)
+      val publicKey = jwk.getPublicKey
+      publicKey match {
+        case rsa: RSAPublicKey => RSA256Verifier(rsa, jwtTimestampLeeway)
+        case ec: ECPublicKey if ec.getParams.getCurve.getField.getFieldSize == 256 =>
+          ECDSAVerifier(Algorithm.ECDSA256(ec, null), jwtTimestampLeeway)
+        case ec: ECPublicKey if ec.getParams.getCurve.getField.getFieldSize == 521 =>
```

Comment:
```suggestion
        case ec: ECPublicKey if ec.getParams.getCurve.getField.getFieldSize == 512 =>
```
typo?

Reply:
if so should not this have been caught by the tests?

Reply:
hehe, no. But you are very attentive
https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/dev/src/Microsoft.IdentityModel.Tokens/JsonWebKeyECTypes.cs#L40

Reply:
oh okay